### PR TITLE
Adds CircleCI 

### PR DESCRIPTION
**Pull request by whereismyjetpack**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,60 +1,43 @@
...
 
-# Build targets for testing
-FROM base as rspec
-CMD [ "/app/bin/ci-rspec" ]
+# Final Target
+FROM base as production
 
-FROM base as eslint
-CMD [ "/app/bin/ci-eslint" ]
+RUN RAILS_ENV=production \
```
Comment (awead) [inline Dockerfile]: @whereismyjetpack this is only for the build that deploys, not the one that's used for running rspec, right? RAILS_ENV would need to be "test" for the build that runs rspec.

File: docker-compose.test.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+version: '3.5' 
+services: 
+  test:
+    environment:
+      REDIS_HOST: redis
+      REDIS_PASSWORD: redispassword
+      OAUTH_APP_ID: ${OAUTH_APP_ID}
+      DRONE_COMMIT_SHA: FOO
```
Comment (awead) [inline docker-compose.test.yml]: @whereismyjetpack should this be there?

File: docker-compose.test.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      OAUTH_APP_URL: ${OAUTH_APP_URL}
+      POSTGRES_DB: scholarsphere
+      POSTGRES_USER: scholarsphere
+      POSTGRES_PASSWORD: scholarsphere
+      AWS_ACCESS_KEY_ID: ${AWS_ACCESS_KEY_ID:-scholarsphere}
+      AWS_SECRET_ACCESS_KEY: ${AWS_SECRET_ACCESS_KEY:-scholarsphere}
+      AWS_BUCKET: scholarsphere
+      AWS_REGION: us-east-1
+      S3_ENDPOINT: http://minio:9000
+      RAILS_ENV: ${RAILS_ENV:-development}
```
Comment (awead) [inline docker-compose.test.yml]: @whereismyjetpack I'm assuming this is the test build for running rspec. Wouldn't you want to default to "test" here?

File: docker-compose.yml [hunk 4]
```diff
@@ -4,7 +4,11 @@ x-web_env: &web_env
       REDIS_HOST: redis
       REDIS_PASSWORD: redispassword
       OAUTH_APP_ID: ${OAUTH_APP_ID}
+      DRONE_COMMIT_SHA: FOO
```
Comment (awead) [inline docker-compose.yml]: @whereismyjetpack see my previous comment about this variable.

File: docker-compose.yml [hunk 5]
```diff
@@ -74,7 +82,7 @@ services:
     volumes:
     - solr-data:/var/solr
     ports:
-    - ${SOLR_PORT:-8983}:8983
+    - 8983:8983
```
Comment (awead) [inline docker-compose.yml]: @whereismyjetpack don't we want to revert this change so we can change the port easily?

File: spec/rails_helper.rb [hunk 6]
```diff
@@ -5,7 +5,7 @@
 ENV['RAILS_ENV'] ||= 'test'
 
 def ci_build?
-  ENV.key?('DRONE_COMMIT_SHA')
+  ENV.key?('DRONE_COMMIT_SHA') || ENV.key?('CIRCLECI') || ENV.key?('CI')
```
Comment (awead) [inline spec/rails_helper.rb]: @whereismyjetpack take out the drone env?

File: Dockerfile (hunk 1 above)
Comment (whereismyjetpack) [reply Dockerfile]: for tests we only run the `base` target. the last target is to bake the assets, and ship the image!

File: docker-compose.test.yml (hunk 2 above)
Comment (whereismyjetpack) [reply docker-compose.test.yml]: we can certainly clean these out.

File: docker-compose.test.yml (hunk 3 above)
Comment (whereismyjetpack) [reply docker-compose.test.yml]: yup. good eye.

File: docker-compose.yml (hunk 5 above)
Comment (whereismyjetpack) [reply docker-compose.yml]: doh!
